aea/configurations/pypi.py

```diff
@@ -235,25 +230,47 @@ def merge_dependencies(dep1: Dependencies, dep2: Dependencies) -> Dependencies:
     """
     Merge two groups of dependencies.
 
-    If some of them are not "simple" (see above), we just filter them out.
+    If some of them are not "simple" (see above), and there is no risk
+      of conflict because there is no other package with the same name,
+      we leave them; otherwise we raise an error.
 
     :param dep1: the first operand
     :param dep2: the second operand.
     :return: the merged dependencies.
     """
     result: Dependencies
-    result = {pkg_name: info for pkg_name, info in dep1.items() if is_simple_dep(info)}
```

Comment:
here we shouldn't ignore 'simple' dependencies, as if a dependency name is present in both operands and one of them is not 'simple', they cannot be merged